### PR TITLE
[kjobctl] Add documentation for Slurm mode

### LINKED ISSUE #2954: [kjobctl] Create a page dedicated to how to use the Slurm simulation.
**What would you like to be added**:
As we discussed [here](https://github.com/kubernetes-sigs/kueue/pull/2825#discussion_r1739187299), we need to have a page dedicated to how to use the Slurm simulation and what caveats to take into account.

**Why is this needed**:

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Adds documentation for the Slurm mode of Kjobctl.

#### Which issue(s) this PR fixes:

Fixes #2954 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+## Before You Begin
+
+Ensure the following prerequisites are met:
+
+- A Kubernetes cluster is operational.
+- The [kjobctl](installation.md) plugin is installed.
+
+## Slurm Mode in Kjob
+
+This mode is specifically designed to support Slurm, a highly scalable cluster management and job scheduling system. Although Kjob does not directly incorporate Slurm's components, it simulates Slurm behavior to facilitate a smooth transition for users. By translating Slurm concepts such as nodes into Kubernetes-compatible terms like pods, Kjob ensures that users familiar with Slurm will find the Kjob environment intuitive and easy to use. The addition of the Slurm mode to the ApplicationProfile in Kjob aims to replicate the experience of using the sbatch command in Slurm as closely as possible, thereby enhancing user satisfaction and efficiency in managing cluster jobs within a Kubernetes context.
```
Comment (alculquicondor) [inline cmd/experimental/kjobctl/docs/run_slurm.md:14]:
```suggestion
This mode is specifically designed to offer a compatibility mode for [Slurm](https://slurm.schedmd.com/) jobs. Kjob does not directly incorporate Slurm's components.
Instead, it simulates Slurm behavior to facilitate a smooth transition for users.
By translating Slurm concepts such as nodes into Kubernetes-compatible terms like pods, Kjob ensures that users familiar with Slurm will find the Kjob environment intuitive and easy to use.
The addition of the Slurm mode to the ApplicationProfile in Kjob aims to replicate the experience of using the sbatch command in Slurm as closely as possible, thereby enhancing user satisfaction and efficiency in managing cluster jobs within a Kubernetes context.
```

In general, please try to split paragraphs in multiple lines, so it's easier to comment.

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+| --gpus-per-task     | Specifies how many GPUs a container inside a pod requires. |
+| -i, --input         | Specifies what to pipe into the script. |
+| -J, --job-name=<jobname> | Specifies the job name. |
+| --mem-per-cpu       | Specifies how much memory a container requires, multiplying the number of requested CPUs per task by mem-per-cpu. |
+| --mem-per-gpu       | Specifies how much memory a container requires, multiplying the number of requested GPUs per task by mem-per-gpu. |
+| --mem-per-task      | Specifies how much memory a container requires. |
+| -N, --nodes         | Specifies the number of pods to be used at a time - parallelism in indexed jobs. |
+| -n, --ntasks        | Specifies the number of identical containers inside of a pod, usually 1. |
+| -o, --output        | Specifies where to redirect the standard output stream of a task. If not passed, it proceeds to stdout and is available via `kubectl logs`. |
+| --partition         | Specifies the local queue name. |
```
Comment (alculquicondor) [inline cmd/experimental/kjobctl/docs/run_slurm.md:34]: link to the Kueue concept page

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        restartPolicy: OnFailure
+```
+
+Once you have created your template, you need an `ApplicationProfile` containing `Slurm` and your new template in the supported modes as shown below:
+
+```yaml
+apiVersion: kjobctl.x-k8s.io/v1alpha1
+kind: ApplicationProfile
+metadata:
+  name: sample-profile
```
Comment (alculquicondor) [inline cmd/experimental/kjobctl/docs/run_slurm.md:104]:
```suggestion
  name: slurm-profile
```

**Review by alculquicondor — COMMENTED** [review]
/assign @mwielgus

File: cmd/experimental/kjobctl/docs/run_slurm.md (hunk 1 above)
Comment (IrvingMg) [reply cmd/experimental/kjobctl/docs/run_slurm.md]: Got it. Thanks

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+      requiredFlags: []
+```
+
+Then, create the ApplicationProfile by running:
+
+```bash
+kubectl create -f config/samples/application_profile.yaml
+```
+
+### 2. Load the Python Script
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/docs/run_slurm.md:127]: @mwielgus WDYT?

[the remark this answers is not in the document]
Comment (mwielgus) [reply cmd/experimental/kjobctl/docs/run_slurm.md:127]: Let's separate the steps. First create a volume with the script but mention that in real life it should be a shared file system (nfs or alike) and using config map is only for illustration purposes.

Then mention volume bundle.

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+| $SLURM_ARRAY_JOB_ID          | Job array’s master job ID number. For now, same as $SLURM_JOB_ID. |
+| $SLURM_ARRAY_TASK_COUNT      | Total number of tasks in a job array. |
+| $SLURM_ARRAY_TASK_MAX        | Job array’s maximum ID (index) number. |
+| $SLURM_ARRAY_TASK_MIN        | Job array’s minimum ID (index) number. |
+
+## Example
+
+The following example demonstrates a use case for running a Python script.
+
+See [config/samples](../config/samples) for the full samples used in this section.
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/docs/run_slurm.md]: Done.

File: cmd/experimental/kjobctl/docs/run_slurm.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/docs/run_slurm.md]: Done, thanks!

File: cmd/experimental/kjobctl/docs/run_slurm.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+### 2. Prepare the Python Script for the Job
+
+Before loading the Python script for use with the pods, it's crucial to consider how the script will be made accessible. 
+
+For the purposes of this tutorial, we'll utilize a ConfigMap to store and pass the Python script. While this method simplifies the setup and is suitable for demonstration purposes, for a scalable setup, using NFS or a similar shared file system is necessary.
+
+#### Creating a ConfigMap with Your Python Script
+
+Create a file named `sample_code.py` and add your Python script. For example:
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/docs/run_slurm.md:135]: Yes, that was added already here, in line 131: https://github.com/kubernetes-sigs/kueue/blob/af6f4512162217b07b6df7c6bfe969b59d042efd/cmd/experimental/kjobctl/docs/run_slurm.md?plain=1#L131

[the remark this answers is not in the document]
Comment (mwielgus) [reply cmd/experimental/kjobctl/docs/run_slurm.md:135]: Ahh, ok

**Review by mimowo — COMMENTED** [review]
/approve
Thanks!

**Pull request merged**